src/core/lib/lattice/hal/default/dcrtpoly.cpp

```diff
@@ -1707,6 +1707,101 @@ void DCRTPolyImpl<VecType>::ExpandCRTBasis(
     this->m_params = paramsQP;
 }
 
+template <typename VecType>
+void DCRTPolyImpl<VecType>::FastExpandCRTBasisPloverQ(
```

Comment:
way too many parameters

Reply:
will pass in CryptoParams instead

Reply:
Make a structure with these parameters

Reply:
Seems like we could create a single structure to include all input parameters for FastExpandCRTBasisPloverQ(), ExpandCRTBasis(), FastBaseConvqToBskMontgomery(), etc.